src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -98,7 +98,10 @@ protected function extractAttributes(object $object, ?string $format = null, arr
             $name = $reflMethod->name;
             $attributeName = null;
 
-            if (str_starts_with($name, 'get') || str_starts_with($name, 'has') || str_starts_with($name, 'can')) {
+            if (
+                (str_starts_with($name, 'get') || str_starts_with($name, 'has') || str_starts_with($name, 'can'))
+                && ctype_upper($name[3] ?? '')
```

Comment:
As you said in the description methods are case insensitive in PHP, so this looks too brittle to me.

Reply:
I barely see any other option that won't imply having a list of exclusions. If you have any better idea, share please.

Reply:
@fabpot?

Reply:
I think this would work for me. It'd just also allow `_` next to the prefix so that we don't break snake cased methods.

Reply:
Actually it might be more robust to ch check with `!ctype_lower`

Reply:
@nicolas-grekas why?

Reply:
because it would account for methods named get_foo or get123, while the current logic doesn't